src/rate-limit.coffee

```diff
@@ -61,16 +61,18 @@ module.exports = (robot) ->
          lastExecutedTime[listenerAndRoom] > Date.now() - minPeriodMs
         # Command is being executed too quickly!
         robot.logger.debug "Rate limiting " + listenerID + " in " + roomOrUser + "; #{minPeriodMs} > #{Date.now() - lastExecutedTime[listenerAndRoom]}"
-        # Notify at least once per rate limiting event
-        myNotifyPeriodMs = minPeriodMs if notifyPeriodMs > minPeriodMs
-        # If no notification sent recently
-        if (lastNotifiedTime.hasOwnProperty(listenerAndRoom) and
-            lastNotifiedTime[listenerAndRoom] < Date.now() - myNotifyPeriodMs) or
-           not lastNotifiedTime.hasOwnProperty(listenerAndRoom)
-          context.response.reply "Rate limit hit! Please wait #{minPeriodMs/1000} seconds before trying again."
-          lastNotifiedTime[listenerAndRoom] = Date.now()
-        # Bypass executing the listener callback
-        done()
+
+        if process.env.HUBOT_RATE_LIMIT_NOTIFY_MSG?
+          # Notify at least once per rate limiting event
+          myNotifyPeriodMs = minPeriodMs if notifyPeriodMs > minPeriodMs
+          # If no notification sent recently
+          if (lastNotifiedTime.hasOwnProperty(listenerAndRoom) and
+              lastNotifiedTime[listenerAndRoom] < Date.now() - myNotifyPeriodMs) or
+             not lastNotifiedTime.hasOwnProperty(listenerAndRoom)
+            context.response.reply "Rate limit hit! Please wait #{minPeriodMs/1000} seconds before trying again."
+            lastNotifiedTime[listenerAndRoom] = Date.now()
+          # Bypass executing the listener callback
+          done()
```

Comment:
This needs to be outside the conditional (it has nothing to do with the notification messages).